packages/driver/src/cypress/cy.ts

```diff
@@ -505,19 +526,57 @@ export class $Cy extends EventEmitter2 implements ITimeouts, IStability, IAssert
         // because they would have been automatically applied during
         // onBeforeAppWindowLoad, but in the case where we visited
         // about:blank in a visit, we do need these
-        this.contentWindowListeners(getContentWindow($autIframe))
+        this.contentWindowListeners(autWindow)
 
-        this.Cypress.action('app:window:load', this.state('window'))
+        // stability is signalled after the window:load event to give event
+        // listeners time to be invoked prior to moving on, but not if
+        // there is a cross-origin error and the cy.origin API is
+        // not utilized
+        try {
+          this.Cypress.action('app:window:load', this.state('window'))
+          const remoteLocation = this.getRemoteLocation()
+
+          cy.state('autOrigin', remoteLocation.originPolicy)
+          this.Cypress.primaryOriginCommunicator.toAllSpecBridges('window:load', { url: remoteLocation.href })
+
+          signalStable()
+        } catch (err: any) {
+          // this catches errors thrown by user-registered event handlers
+          // for `window:load`. this is used in the `catch` below so they
+          // aren't mistaken as cross-origin errors
+          err.isFromWindowLoadEvent = true
+
+          signalStable()
+
+          throw err
+        }
```

Comment:
```suggestion
        } finally {
          signalStable()
        }
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21137/commits/348302afd76fb3a1c18d953c50acf544165a50f9